src/NuGet.Protocol.Core.v3/DownloadResourceV3.cs

```diff
@@ -18,33 +18,81 @@ public class DownloadResourceV3 : DownloadResource
     {
         private readonly RegistrationResourceV3 _regResource;
         private readonly HttpClient _client;
+        private readonly string _flatContainerBaseUrl;
 
+        /// <summary>
+        /// Download packages using the download url found in the registration resource.
+        /// </summary>
         public DownloadResourceV3(HttpClient client, RegistrationResourceV3 regResource)
+            : this(client)
         {
             if (client == null)
             {
-                throw new ArgumentNullException("client");
+                throw new ArgumentNullException(nameof(regResource));
```

Comment:
Looks like you are checking if client is null and throw ArgNull for regResource. Perhaps, you need 2 of them